api/wasm/v0/cpp/proxy_wasm_externs.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Proxy-WASM ABI.
+ */
+// NOLINT(namespace-envoy)
+
+#pragma once
+
+#include <stddef.h>
+#include <stdint.h>
+
+//
+//  ABI calls from the host into the VM.
+//
+
+// Non-stream calls.
+extern "C" uint32_t proxy_on_start(uint32_t root_context_id, uint32_t configuration_size);
```

Comment:
Can you write full Doxygen comments for these in the style of Envoy's `include` interfaces? It's unclear how these are to be used, the preconditions, state machine, what the args are etc.

Reply:
I can do that, but we have an ABI document which @PiotrSikora wrote which might be a better way to document these functions rather than just as Doxygen.  In that document we can describe the lifecycle, use illustrations etc.   Should we reference such documentation here or would you like both Doxygen style comments in addition to a complete ABI document ?

Reply:
Envoy's interface headers are canonical for this, so I think we should have them here. It's fine to iterate in the ABI document, but ultimately the docs need to be in the source.

Reply:
Do you want me to add the full set of ABI calls at this time or just this reduced set that I pulled out for the initial drop with tests ?

Reply:
Let's just start with the reduced set so we can agree on general conventions. This will make everything else go much faster.

Reply:
Added doxygen comments. PTAL